src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.SingleLine.vb

```diff
@@ -0,0 +1,84 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InvertIf
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.InvertIf), [Shared]>
+    Friend NotInheritable Class VisualBasicInvertSingleLineIfCodeRefactoringProvider
+        Inherits VisualBasicInvertIfCodeRefactoringProvider(Of SingleLineIfStatementSyntax)
+
+        Protected Overrides Function GetHeaderSpan(ifNode As SingleLineIfStatementSyntax) As TextSpan
+            Return TextSpan.FromBounds(
+                    ifNode.IfKeyword.SpanStart,
+                    ifNode.Condition.Span.End)
+        End Function
+
+        Protected Overrides Function IsElseless(ifNode As SingleLineIfStatementSyntax) As Boolean
+            Return ifNode.ElseClause Is Nothing
+        End Function
+
+        Protected Overrides Function CanInvert(ifNode As SingleLineIfStatementSyntax) As Boolean
+            Return TypeOf ifNode.Parent IsNot SingleLineLambdaExpressionSyntax AndAlso
+                Not ifNode.Statements.Any(Function(n) n.IsKind(SyntaxKind.LocalDeclarationStatement)) AndAlso
```

Comment:
This appears to be causing some VB tests to no longer offer the refactoring where it previously did:
TestEscapeKeywordsIfNeeded1
TestEscapeKeywordsIfNeeded2
TestEscapeKeywordsIfNeeded3
TestEscapeKeywordsIfNeeded5
TestAddEmptyArgumentListIfNeeded
TestParenthesizeToKeepParseTheSame2

Reply:
@chborl There was some discussion during this whole PR about the complexity necessary to support such extremely esoteric cases.  Basically, it's super unlikely to hit those cases in pratice (you'd need a query, in a single-line if, followed by a specific expression with a certain name).  Trying to support this case just seemed like far too much work and complexity to warrant inclusion.

I think i suggested just scrapping that code and saying: if we run into this situation, just don't even bother offering the feature.  It helps make things much simpler, and allows us to focus on providing much more real value in teh feature by supporting real cases that actually will be hit far more often.

Now, if we want, we can take this for a design-change signoff.  However, IMO, this is completely reasonable and worth it :)

Reply:
@chborl Apparently we discussed this together in the past:

https://github.com/dotnet/roslyn/pull/26793#discussion_r187782708

> note: as per my conversation with @chborl i think the case that VB is handling is incredibly esoteric. I would be ok with having VB just checking for that case and not offerring invert-if if it helps simplify the rest of the algorithms here. I think it's a practically irrelevant case in practice and i would be fine with no invert-if for it.

This is an unfortunate consequence of having PRs drag on for so long.  We literally forget about the conversations we've had on these very topics. :)

Reply:
Note: i still stand by that statement.  We should not be adding lots of complexity to features to handle a 0.0001% case.  We should be investing our efforts in the common cases that proviee high value, and adding support for things people will be doing all the time.  In htat regard, this PR is really helping out this feature.  It goes from being a basic "i only support inversion in this one basic case" to "i support so many common flow control patterns".  We shoudl not make that job much more difficult to just to support some esoteric cases that were literally dreamt up by a genius dev who loved finding these sorts of crazy cases for fun, but which don't actually come in in the real world :)